source/index.md

```diff
@@ -68,6 +68,28 @@ The following illustrate of what can be achieved with the Nix ecosystem:
 - Atomic upgrades and rollbacks.
 
 
+## Who is Nix for?
+
+Nix is primarily a tool for software developers.
```

Comment:
Shouldn't this be a PR to nixos.org instead? nix.dev is for learning about Nix, but this section doesn't teach anything, it's just marketing. If somebody went onto nix.dev, we should assume that they want to learn about Nix, so I don't think this is necessary or useful here.

And if your intention is to make Nix platform _contributors_ be more focused on improving the developer story, I also don't think this would achieve that. Nix is a community of various individuals and companies, which will do whatever they're most interested in (which often isn't focused on the developer story).

Reply:
> Shouldn't this be a PR to nixos.org instead?

In an ideal world, maybe? In practice, I find that code unmaintainable and refuse to work with it.

> nix.dev is for learning about Nix, but this section doesn't teach anything, it's just marketing.

This is very clearly not marketing, because it doesn't advertise anything, it doesn't try to convince anyone of anything. The purpose is, as said in the commit message, to set expectations and prepare people for what they'll deal with. The phrase "going back to a world without Nix" is only a measure to avoid sounding academic. The technically correct statement would be "According to community surveys 2022 and 2023, people who identify with these descriptions very probably find Nix useful".

> And if your intention is to make Nix platform contributors be more focused on improving the developer story, I also don't think this would achieve that.

While that's not my intention, I strongly doubt it's true what you say. [The first impression matters a lot](https://en.wikipedia.org/wiki/Priming_(psychology)), and even if this particular piece is below the fold, whoever cares enough to read it will carry that message around with them for a long time to shape their thinking.

Reply:
> This is very clearly not marketing

Fair yeah, but still, setting expectations seems like a main landing-page kind of deal. And you refusing to touch nixos.org is clearly something gone wrong that we should prioritise addressing instead. This is definitely worth a @NixOS/marketing-team ping.

Reply:
I guess it shouldn't hurt to _also_ set expectations for nix.dev specifically, but so far this PR seems to set expectations for Nix in general.

Reply:
I'm being deliberately generic here, because we don't have that sort of writing anywhere. I'd rather have it here now and move it around later. I've been repeating myself and trying to come up with things that help orient people for almost two years now, getting bogged down with concerns on every step. I'm all in favor of being considerate with far-reaching decisions, but *come on!*

Reply:
I'm all in favor of getting more content written as well! For now, I'd rather have redundant information in different places than try to keep it in a single place (as long as that information won't go out of date soon). It's been my experience that people find nix through multiple different links, and often don't visit all the other links (e.g. they find combinations of the nix pills, manuals, nix.dev, wiki, and multiple other sites, but often don't find all of those), so having repeating information (which doesn't go out of date soon) is better IMO at the moment.

Properly organising all the content under a single structure takes way more time (but we're moving there, e.g. with the nix.dev -> docs.nixos.org rename and so on), and when we get there we can start cutting the redundant content.

My only concern is making sure what we're adding is:
1. Not confusing for the intended audience
2. Improving on the current content

I think the latest version of this PR fits both criteria, so I'm approving!